devito/types/grid.py

```diff
@@ -127,21 +129,23 @@ def __init__(self, shape, extent=None, origin=None, dimensions=None,
                                      distributor=self._distributor, counter=counter)
         self._subdomains = subdomains
 
-        origin = as_tuple(origin or tuple(0. for _ in self.shape))
-        self._origin = tuple(self._const(name='o_%s' % d.name, value=v, dtype=self.dtype)
-                             for d, v in zip(self.dimensions, origin))
+        self._origin = as_tuple(origin or tuple(0. for _ in self.shape))
+        self._origin_symbols = tuple(Scalar(name='o_%s' % d.name, dtype=dtype,
+                                            is_const=True)
+                                     for d in self.dimensions)
 
         # Sanity check
         assert (self.dim == len(self.origin) == len(self.extent) == len(self.spacing))
 
         # Store or create default symbols for time and stepping dimensions
         if time_dimension is None:
-            spacing = self._const(name='dt', dtype=self.dtype)
-            self._time_dim = self._make_time_dim(spacing)
-            self._stepping_dim = self._make_stepping_dim(self.time_dim, name='t')
+            spacing = Scalar(name='dt', dtype=dtype, is_const=True)
```

Comment:
I would still have a default value (dt=1) similar to how we have a default value for the extent. But not sure if breaks stuff.

Reply:
I'm all but convinced about this one. I think it could lead to various user level errors

Reply:
Ok, fine with me.